### PR TITLE
fix(payments): add checkout route to INDEX_ROUTES

### LINKED ISSUE #10041: Navigating to the new checkout route on stage returns a 404
See this [slack thread](https://mozilla.slack.com/archives/C4D36CAJW/p1628175054008600). Per Barry, this is because the new route isn’t added to `INDEX_ROUTES` in fxa-payments-server/server/lib/server.js.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/FXA-3826)

### PR DESCRIPTION
**Pull request by biancadanforth**
Because:

* We are seeing a 404 on stage for the new checkout route

This commit:

* Ensures we correctly route to index.html for this new route.

Closes #10041

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

Regression introduced by #9793

### CODE REVIEW

File: packages/fxa-payments-server/server/lib/server.js [hunk 1]
```diff
@@ -107,6 +107,7 @@ module.exports = () => {
     '/v1/products/:productId',
     '/v2/subscriptions',
     '/v2/products/:productId',
+    '/v1/checkout/:productId',
```
Comment (chenba) [inline packages/fxa-payments-server/server/lib/server.js:110]: I don't think we need the '/v1' part.

**Review by chenba — CHANGES_REQUESTED** [review]
'/v1' needn't be part of the route.

File: packages/fxa-payments-server/server/lib/server.js (hunk 1 above)
Comment (biancadanforth) [reply packages/fxa-payments-server/server/lib/server.js:110]: Admittedly I was cargoculting -- but why is it there for the other routes?

Comment (biancadanforth) [reply packages/fxa-payments-server/server/lib/server.js:110]: I guess I see without, with a v1 and a v2.

**Review by chenba — APPROVED** [review]

File: packages/fxa-payments-server/server/lib/server.js (hunk 1 above)
Comment (chenba) [reply packages/fxa-payments-server/server/lib/server.js]: Yeah, for a while we needed all those.  It should be safe to clean out the v1 and v2 stuff at this point.

Comment (biancadanforth) [reply packages/fxa-payments-server/server/lib/server.js]: I'll do that here while I'm in here.

**Comment by biancadanforth** [comment]
Latest one removes all the `/v1` and `/v2` routes.

**Pull request merged**